presubmit-service/app.ts

```diff
@@ -0,0 +1,58 @@
+import * as express from 'express';
+import * as shell from 'shelljs';
+import * as bodyParser from 'body-parser';
+
+interface CheckPrRequest {
+  pr: string;
+  repo: string;
+  org: string;
+  branch: string;
+}
+
+/** The port to listen for HTTP requests on. */
+const PORT = process.env.PORT || 8080;
+/** The express application instance. */
+const APP = express();
+
+// Since POST requests send all data in the body, it needs to be parsed as a
+// JSON object to be easily accessed.
+APP.use(bodyParser.json());
+
+// Handle POST requests at /check_pr.
+APP.post('/check_pr', (req: express.Request, res: express.Response) => {
+  const options = req.body as CheckPrRequest;
+  const {pr, repo, org, branch} = options;
+
+  // All responses should be JSON responses.
+  res.setHeader('content-type', 'application/json');
+  // Using the Cache-Control header, prevent caching of requests.
+  res.setHeader('Cache-Control', 'private, no-cache, no-store, must-revalidate');
+
+  // Check if all parameters are provided.
+  if (!pr || !repo || !branch || !org) {
+    res.status(400);
+    res.send({status: 'missing_params', message: `Missing required parameter(s)`});
+    return;
+  }
+  
+  // Determine if the PR can apply to the repos targeted branch and return the result.
+  const prCheck = shell.exec(`./check_if_pr_can_cherry_pick.sh ${org} ${repo} ${pr} ${branch}`);
+  if (prCheck.code === 0) {
```

Comment:
I would break the `shell.exec` and the response code check into its own function like `canCheryPick`, probably in another file (keeping this file focused purely on the request/response handling).